### DISCUSSION TITLE
trim galore tutorial

### DISCUSSION THREAD

**Discussion by meganewing** [sr320/course-fish546-2021#116] [Q&A] [answered]
I was wondering if theres time during class tomorrow if you could do a trim galore tutorial? I've watched a few online but am still a little confused in some places

**Comment by sr320** [comment]
Can you be a bit more specific about what places are confusing?

**Reply by meganewing** [reply]
Im not sure how to specify which files I want trimmed. When watching a tutorial last night they used the cutadapt command and then specified the adapter sequence but not the file they're trimming. 
I also don't know what the illumina sequence is that I'm trimming.

**Reply by kubu4** [reply]
I'd say ignore the tutorial and, instead, read the manual: https://github.com/FelixKrueger/TrimGalore/blob/master/Docs/Trim_Galore_User_Guide.md

> I also don't know what the illumina sequence is that I'm trimming.

Virtually all trimming software (TrimGalore included) automatically identifies Illumina sequencing adapters. They also offer the user the ability to supply custom sequences that you'd like trimmed.

TrimGalore does have one annoying caveat that can cause issues if you're unfamiliar with the software. This caveat is that it will identify the most abundant set of sequences that correspond to a know Illumina adaptor and will trim _only_ that, even if you have multiple sequencing adaptors (e.g. paired-end sequencing). That means, the user frequently has to run TrimGalore a _second_ time in order to ensure all adaptor sequences are trimmed. Other trimming software doesn't seem to have this caveat.

**Comment by kubu4** [comment, marked as answer]
I'll also put in a vote/recommendation for using `fastp` instead. It's faster and automatically generates pre-/post-trimming graphs, similar to FastQC. Plus, it's compatible with MultiQC.
